web/src/containers/activity/Goals.js

```diff
@@ -9,66 +9,149 @@ import {
+    const { goals: data } = props;
+    const lastKey = data.length ? data[data.length - 1].key : null;
+
+    if (lastKey && !(lastKey in state.showForm)) {
+      return {
+        showForm: {
+          ...arrToObj(Object.keys(state.showForm), false),
+          [lastKey]: true
+        }
+      };
+    }
+
+    return null;
+  }
+
+  constructor(props) {
+    super(props);
+
+    const goalKeys = props.goals.map(e => e.key);
+
+    const showForm = goalKeys.reduce(
+      (obj, key, i) => ({ ...obj, [key]: i === 0 }),
+      {}
+    );
+
+    this.state = { showForm };
+  }
+
   handleSync = (index, field) => html => {
-    const { activity, updateActivity } = this.props;
+    const { activityKey, updateActivity } = this.props;
     const updates = { goals: { [index]: { [field]: html } } };
-    updateActivity(activity.key, updates);
+    updateActivity(activityKey, updates);
   };
 
   handleDelete = key => () => {
-    const { activity, removeActivityGoal } = this.props;
-    removeActivityGoal(activity.key, key);
+    const { activityKey, removeActivityGoal } = this.props;
+    removeActivityGoal(activityKey, key);
   };
 
   handleAdd = () => {
-    const { activity, addActivityGoal } = this.props;
-    addActivityGoal(activity.key);
+    const { activityKey, addActivityGoal } = this.props;
+    addActivityGoal(activityKey);
+  };
+
+  toggleForm = expenseKey => () => {
+    this.setState(prev => ({
+      showForm: {
+        ...prev.showForm,
+        [expenseKey]: !prev.showForm[expenseKey]
+      }
+    }));
   };
 
   render() {
-    const { activity } = this.props;
+    const { goals } = this.props;
+    const { showForm } = this.state;
 
     return (
       <Subsection resource="activities.goals" nested>
-        {activity.goals.map((d, i) => (
-          <div key={d.key} className="mb3">
-            <SubsectionChunk resource="activities.goals.goal">
-              <div className="mb3">
-                <div className="mb-tiny">
-                  {activity.goals.length > 1 && (
-                    <Btn
-                      kind="outline"
-                      extraCss="right px1 py-tiny h6 line-height-1"
-                      onClick={this.handleDelete(d.key)}
-                    >
-                      Remove
-                    </Btn>
-                  )}
-                  {t('activities.goals.goal.title', { number: i + 1 })}
-                </div>
-                <RichText
-                  content={d.description}
-                  onSync={this.handleSync(i, 'description')}
+        {goals.length === 0 ? (
+          <NoDataMsg>{t('activities.expenses.noDataNotice')}</NoDataMsg>
+        ) : (
+          <div className="mt3 pt3 border-top border-grey">
+            {goals.map((goal, i) => (
+              <div key={goal.key}>
+                <MiniHeader
+                  content={
+                    <div className="col-11 truncate">
+                      {plaintext(goal.description)}
```

Comment:
what are your thoughts on using this `plaintext` method vs. the `Md` component here?

Reply:
What would happen if the start of the description was an H1?  I'm open to giving it a try and seeing what happens - the `plaintext` method doesn't exactly fill me with joy.  😆

Reply:
ohh, that's a good point -- alright, plaintext works for me then!